datadog_lambda/span_pointers.py

```diff
@@ -89,3 +95,64 @@ def _calculate_s3_span_pointers_for_object_created_s3_information(
             str(e),
         )
         return []
+
+
+def _calculate_dynamodb_span_pointers_for_event(event) -> List[_SpanPointerDescription]:
+    # Example event:
+    # https://docs.aws.amazon.com/lambda/latest/dg/with-ddb.html
+
+    return list(
+        chain.from_iterable(
+            _calculate_dynamodb_span_pointers_for_event_record(record)
+            for record in event.get("Records", [])
+        )
+    )
+
+
+def _calculate_dynamodb_span_pointers_for_event_record(
+    record,
+) -> List[_SpanPointerDescription]:
+    try:
+        table_name = _extract_table_name_from_dynamodb_stream_record(record)
+        primary_key = record["dynamodb"]["Keys"]
+
+    except Exception as e:
+        logger.warning(
+            "missing DynamoDB information required to make a span pointer: %s",
+            str(e),
+        )
+        return []
+
+    try:
+        return [
+            _aws_dynamodb_item_span_pointer_description(
+                pointer_direction=_SpanPointerDirection.UPSTREAM,
+                table_name=table_name,
+                primary_key=primary_key,
+            )
+        ]
+
+    except Exception as e:
+        logger.warning(
+            "failed to generate DynamoDB span pointer: %s",
+            str(e),
+        )
+        return []
+
+
+def _extract_table_name_from_dynamodb_stream_record(record) -> str:
+    # Example eventSourceARN:
+    # arn:aws:dynamodb:us-east-2:123456789012:table/my-table/stream/2024-06-10T19:26:16.525
+    event_source_arn = record["eventSourceARN"]
+
+    [_arn, _aws, _dynamodb, _region, _account, dynamodb_info] = event_source_arn.split(
+        ":", maxsplit=5
+    )
+    if _arn != "arn" or _aws != "aws" or _dynamodb != "dynamodb":
+        raise ValueError(f"unexpected eventSourceARN format: {event_source_arn}")
```

Comment:
Do we need to check all 3? Could we just check `if _arn != "arn":`? Just for performance sake

Reply:
we don't have to check all three, but i prefer to be super cautious here.